concent_api/core/subtask_helpers.py

```diff
@@ -150,12 +152,36 @@ def update_subtask_state(
 
 def are_keys_and_addresses_unique_in_message_subtask_results_accepted(subtask_results_accepted_list: List[SubtaskResultsAccepted]) -> bool:
 
-    unique_public_keys = set(subtask_results_accepted.task_to_compute.requestor_public_key for subtask_results_accepted in subtask_results_accepted_list)
-    unique_ethereum_addresses = set(subtask_results_accepted.task_to_compute.requestor_ethereum_address for subtask_results_accepted in subtask_results_accepted_list)
-    unique_ethereum_public_keys = set(subtask_results_accepted.task_to_compute.requestor_ethereum_public_key for subtask_results_accepted in subtask_results_accepted_list)
+    unique_requestor_public_keys = set(subtask_results_accepted.task_to_compute.requestor_public_key for subtask_results_accepted in subtask_results_accepted_list)
+    unique_requestor_ethereum_addresses = set(subtask_results_accepted.task_to_compute.requestor_ethereum_address for subtask_results_accepted in subtask_results_accepted_list)
+    unique_requestor_ethereum_public_keys = set(subtask_results_accepted.task_to_compute.requestor_ethereum_public_key for subtask_results_accepted in subtask_results_accepted_list)
 
-    is_public_key_unique = (len(unique_public_keys) == 1)
-    is_ethereum_address_unique = (len(unique_ethereum_addresses) == 1)
-    is_ethereum_public_key_unique = (len(unique_ethereum_public_keys) == 1)
+    unique_provider_public_keys = set(subtask_results_accepted.task_to_compute.provider_public_key for subtask_results_accepted in subtask_results_accepted_list)
+    unique_provider_ethereum_addresses = set(subtask_results_accepted.task_to_compute.provider_ethereum_address for subtask_results_accepted in subtask_results_accepted_list)
+    unique_provider_ethereum_public_keys = set(subtask_results_accepted.task_to_compute.provider_ethereum_public_key for subtask_results_accepted in subtask_results_accepted_list)
 
-    return is_public_key_unique and is_ethereum_address_unique and is_ethereum_public_key_unique
+    is_requestor_public_key_unique = (len(unique_requestor_public_keys) == 1)
+    is_requestor_ethereum_address_unique = (len(unique_requestor_ethereum_addresses) == 1)
+    is_requestor_ethereum_public_key_unique = (len(unique_requestor_ethereum_public_keys) == 1)
+
+    is_provider_public_key_unique = (len(unique_provider_public_keys) == 1)
+    is_provider_ethereum_address_unique = (len(unique_provider_ethereum_addresses) == 1)
+    is_provider_ethereum_public_key_unique = (len(unique_provider_ethereum_public_keys) == 1)
+
+    requestor_public_key = subtask_results_accepted_list[0].task_to_compute.requestor_public_key
+    unique_signature = all(
```

Comment:
I would suggest renaming it to: `are_all_messages_signed_by_requestor` or shorter `are_all_signed_by_requestor`, or even shorter `all_signed_by_requestor`

Reply:
Done